### PR TITLE
NZ Superannuation

### PR DESCRIPTION
**Pull request by dlouise64**
- Added NZ Superannuation

### CODE REVIEW

**Comment by Br3nda** [comment]
I think I can review this this afternoon. Can you address the code climate issues?

**Comment by dlouise64** [comment]
@Br3nda all code climate has been fixed, ready for review.

File: openfisca_aotearoa/variables/acts/super/super.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+class super__living_alone(Variable):
+    value_type = int
+    entity = Person
+    definition_period = MONTH
+    label = u"Is a person considered as living alone"
+    reference = "http://www.legislation.govt.nz/act/public/2001/0084/latest/whole.html#DLM5578822"
+
+
+class super__is_receiving_compensation(Variable):
```
Comment (verbman) [inline openfisca_aotearoa/variables/acts/super/super.py]: I wonder if this variable should be called acc__is_receving_compensation in that the reference link is to the ACC act?

File: openfisca_aotearoa/variables/acts/super/super.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = u"Is receiving compensation payment through ACC"
+    reference = "http://www.legislation.govt.nz/act/public/2001/0049/latest/DLM105404.html#DLM105404"
+
+
+class super__has_partner(Variable):
```
Comment (verbman) [inline openfisca_aotearoa/variables/acts/super/super.py]: Perhaps we should call this variable super__has_partner_in_long_term_care_or_rest_home

File: openfisca_aotearoa/variables/acts/super/super.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    label = u"Has a partner"
+    reference = "http://www.legislation.govt.nz/act/public/2001/0084/latest/DLM114223.html"
+
+
+class super__is_veteran(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = u"Applicant is a veteran"
+    reference = "http://www.legislation.govt.nz/act/public/2001/0084/latest/DLM115004.html"
```
Comment (verbman) [inline openfisca_aotearoa/variables/acts/super/super.py]: This reference doesn't appear to link to the appropriate section? That is it links to the entitlements page and which appears disconnected from the is_veteran aspect of the variable name. If a "veteran" isn't defined in the super act we might need a reference to the "Veterans’ Support Act 2014"...

File: openfisca_aotearoa/variables/acts/super/super.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+class super__eligibility(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = u"Classified as eligible for NZ Super"
+
+    def formula(persons, period, parameters):
+        return (persons('age', period) >= 65) *\
+            not_(persons('super__is_receiving_compensation', period)) +\
```
Comment (verbman) [inline openfisca_aotearoa/variables/acts/super/super.py]: Just checking, I would have thought super__has_partner would be referenced in here? Or is that variable only needed for the calculation of the amount?

Comment (dlouise64) [reply openfisca_aotearoa/variables/acts/super/super.py]: @verbman yes you're correct, its only needed for calculation.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+# 5.1.2 - [#94](https://github.com/ServiceInnovationLab/openfisca-aotearoa/pull/94)
+* Added NZ Super
```
Comment (Br3nda) [inline CHANGELOG.md]: Can we have the full name of the act here instead of just NZ super

File: openfisca_aotearoa/variables/acts/super/super.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    definition_period = MONTH
+    label = u"Has a partner in long term care or rest home"
+    reference = "http://www.legislation.govt.nz/act/public/2001/0084/latest/DLM114223.html"
+
+
+class super__eligibility(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = u"Classified as eligible for NZ Super"
```
Comment (verbman) [inline openfisca_aotearoa/variables/acts/super/super.py]: Can we have a reference for the variable?

File: openfisca_aotearoa/variables/acts/super/super.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    def formula(persons, period, parameters):
+        return (persons('age', period) >= 65) *\
+            not_(persons('acc__is_receiving_compensation', period)) +\
+            persons('veterans_support__is_entitled_to_be_paid_veterans_pension', period)
```
Comment (verbman) [inline openfisca_aotearoa/variables/acts/super/super.py]: Is this variable returning true if the person gets the normal pension or the veterans pension?

File: openfisca_aotearoa/variables/acts/immigration/residence.py [hunk 8]
```diff
@@ -56,6 +56,13 @@ class number_of_years_lived_in_nz(Variable):
     label = u"Number of years lived in NZ"
 
 
+class total_number_of_years_lived_in_nz_since_age_20(Variable):
```
Comment (verbman) [inline openfisca_aotearoa/variables/acts/immigration/residence.py]: Lets move this to demographics (it's not defined in the act)

**Pull request merged**